### ISSUE TITLE
Task Area: tabs use incorrect keyboard nav

### ISSUE THREAD

**Issue by shaunanoordin** [zooniverse/front-end-monorepo#7509]
## Accessibility Issue

Package: lib-classifier

The "Task" and "Tutorials" tabs in the Classifier's Task Area uses incorrect keyboard navigation. To switch from one tab to the next, the volunteer currently has to press the `Tab` key; the correct/expected interaction should be pressing the `LeftArrow` key or `RightArrow` key.

_Screenshot: example task area. "Task" tab has keyboard focus. Pressing 'Tab' key switches focus to 'Tutorial' tab_

<img width="364" height="429" alt="Image" src="https://github.com/user-attachments/assets/07e6f205-d219-4092-baa0-d576353c5c47" />

Note: if a workflow doesn't have a tutorial, the tab is visible but disabled and therefore can't receive focus in any way.

### Dev Notes

The problem is with [Grommet's](https://v2.grommet.io/tabs) `<Tabs>` and `<Tab>` component, so it's baked in.  You can confirm this on the [examples](https://v2.grommet.io/tabs) on the Grommet page, and there's [an issue open on their repo](https://github.com/grommet/grommet/issues/7327).

Accessibility documentation:
- MDN: [ARIA `tab` role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/tab_role) -> [keyboard interactions](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/tab_role#keyboard_interactions)

### Status

Low priority.

**Comment by eatyourgreens** [comment]
`focusgroup` is currently behind a feature flag in Chrome (while they solicit developer feedback.) Once it ships in browsers, it will support arrow key navigation by default for grouped controls like tabs.

- https://developer.chrome.com/blog/focusgroup-rfc
- https://open-ui.org/components/scoped-focusgroup.explainer/
- https://www.stevefrenzel.dev/posts/my-thoughts-on-the-focusgroup-attribute-proposal/

EDIT: here's a code example for a tablist using `focusgroup`. Note that focusgroup assigns the appropriate roles and states to the container `div` and also the child buttons. There's no need to add ARIA roles or states.

```html
<div focusgroup="tablist" aria-label="Example tabs">
  <button type="button">Hammer</button>
  <button type="button">Sandwich</button>
  <button type="button">Berry</button>
  <button type="button">Carwash</button>
  <button type="button">Milkbox</button>
</div>
```

**Comment by eatyourgreens** [comment]
In the meantime, there's also the [`roving-tabindex` web component](https://github.com/jakelazaroff/roving-tabindex), which implements roving tab-index for a group of controls:

```html
<roving-tabindex selector="button">
  <ul>
    <li><button>One</button></li>
    <li><button>Two</button></li>
    <li><button>Three</button></li>
  </ul>
</roving-tabindex>
```

**Comment by eatyourgreens** [comment]
>[!NOTE]
`focusgroup` shipped today in Chrome 150. 
>
>https://developer.chrome.com/blog/new-in-chrome-150?hl=en#focusgroup
>
>There's a polyfill for other browsers, and older Chrome. 
>
>https://npmx.dev/@microsoft/focusgroup-polyfill
>
>EDIT: Adrian Roselli has published test code for Chrome 150: https://adrianroselli.com/2026/07/focusgroup-tests.html#Tablist

**Comment by eatyourgreens** [comment]
[WebAIM: Tabbed Interfaces](https://webaim.org/techniques/tabs/) also points out that tab panels should be focusable, so that tabbing from a tab takes you to its panel.

**Comment by eatyourgreens** [comment]
Some other keyboard accessibility bugs I've found, specific to the classifier task tabs:
- Activating the Tutorial tab with Enter/Space opens a popup tutorial for anonymous users, along with the tutorial tab. See #7292.
- Activating the Task tab with Enter/Space loses keyboard focus. I think it's stolen by `autofocus` in the newly mounted task component.

**Comment by eatyourgreens** [comment]
It also turns out that there are three implementations of tabs in the monorepo code. None of them have the expected keyboard interactions for a tabbed interface:
- [`@zooniverse/react-components Tabs`](https://github.com/zooniverse/front-end-monorepo/tree/main/packages/lib-react-components/src/Tabs): Zooniverse-themed Grommet tabs.
- [`@zooniverse/user StatsTabs`](https://github.com/zooniverse/front-end-monorepo/blob/main/packages/lib-user/src/components/UserHome/components/Dashboard/components/StatsTabs/StatsTabs.jsx): Styled Grommet tabs, but not Zooniverse themed (I think.)
- [`@zooniverse/user MainContent`](https://github.com/zooniverse/front-end-monorepo/blob/main/packages/lib-user/src/components/shared/MainContent/MainContent.jsx): bespoke tabbed UI with ARIA roles and states (for screen readers) but no custom keyboard behaviour (so people who rely on screen readers will hear that these are tabs, but then be confused and/or frustrated when they try to use the keyboard to change the selected tab and it doesn't work.)

**Comment by eatyourgreens** [comment]
#7539 fixes this, in the classifier, by adding `focusgroup` to the Zooniverse `Tabs` component.

**Comment by eatyourgreens** [comment]
#7545 applies the same fix to the `MainContent` tabs in `@zooniverse/user`, so that they're usable by people who use screen readers and expect the cursor keys to control tabs.

**Comment by eatyourgreens** [comment]
[Using tabbed interfaces with a screen reader](https://tink.uk/using-tabbed-interfaces-with-a-screen-reader/), by Léonie Watson, points out that a blind user will expect the arrow keys to switch between tabs in a tabbed interface. This issue will block BVI volunteers from using tabbed interfaces on the Zooniverse web site, and raises barriers for sighted keyboard users.

**Comment by eatyourgreens** [comment]
When I use the keyboard to navigate tabbed interfaces on the Zooniverse site, the styling can be ambiguous. Here, Classifications is selected, and underlined. Hours is focused, and also underlined. It's not immediately clear whether the chart is showing classifications or hours.

<img width=800 alt="The MainContent component from @zooniverse/user. The Classifications tab is selected, and underlined. The Hours tab is focused, and also underlined." src="https://github.com/user-attachments/assets/3e719708-002b-4eb0-9851-46edaf788107" />

https://zooniverse.github.io/front-end-monorepo/@zooniverse/user/index.html?path=/story/components-shared-maincontent--default
